server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/RemoteDeliveryErrorHandlingTest.java

```diff
@@ -108,7 +109,7 @@ void setup(DockerMockSmtp dockerMockSmtp) throws Exception {
                         .matcher(IsRemoteDeliveryPermanentError.class)
                         .addProperty("repositoryPath", REMOTE_DELIVERY_PERMANENT_ERROR_REPOSITORY.asString()))
                     .addMailet(MailetConfiguration.builder()
-                        .matcher(All.class)
+                        .matcher(IsRemoteDeliveryTemporaryError.class)
```

Comment:
I don't think that is needed. proof: you re-enabled no tests.

Solution: 

Manage to have a mail getting in the RemoteDelivery but reaching this "by accident", in which case we don't want to store it as a temporary error but maybe in another dedicated mail repository.